### DISCUSSION TITLE
duration_only error after update

### DISCUSSION THREAD

**Discussion by Logiar** [kimai/kimai#4059] [Questions & Answers] [answered]
I updated and got [OK] Congratulations! Successfully updated Kimai to version 2.0.23, but when I try to load in I get this error and a 500 response in the gui
`[2023-05-25T04:29:05.608413+00:00] request.CRITICAL: Uncaught PHP Exception Twig\Error\RuntimeError: "An exception has been thrown during the rendering of a template ("You have requested a non-existent service "duration_only".")." at /opt/kimai/vendor/kevinpapst/tabler-bundle/templates/includes/menu.html.twig line 2 {"exception":"[object] (Twig\\Error\\RuntimeError(code: 0): An exception has been thrown during the rendering of a template (\"You have requested a non-existent service \"duration_only\".\"). at /opt/kimai/vendor/kevinpapst/tabler-bundle/templates/includes/menu.html.twig:2)\n[previous exception] [object] (Symfony\\Component\\DependencyInjection\\Exception\\ServiceNotFoundException(code: 0): You have requested a non-existent service \"duration_only\". at /opt/kimai/src/Timesheet/TrackingModeService.php:44)"} []`
I see the "Time-tracking mode duration_only was removed, existing installations will be switched to duration_fixed_begin" in the upgrade info, but how do I trigger the switch to duration_fixed_begin if it wasn't changed during the update?

**Comment by Logiar** [comment, marked as answer]
Managed to find it in https://github.com/kimai/kimai/blob/main/migrations/Version20230126002050.php, seems to already be in the migration table though. Ran the specific query again which seemed to fix it. Didn't have time before work to check if there were other signs of the migration not having ran properly.

**Reply by kevinpapst** [reply]
That's weird. If you ran `bin/console kimai:update` that should have happened already.
Please check if the other update SQL statements are missing as well, which might lead to unexpected behavior.

**Reply by Logiar** [reply]
Had time to check now. Seems things that change the table has gone through.
```
+------------------+--------------+------+-----+---------+----------------+
| Field            | Type         | Null | Key | Default | Extra          |
+------------------+--------------+------+-----+---------+----------------+
| id               | int(11)      | NO   | PRI | NULL    | auto_increment |
| name             | varchar(60)  | NO   | UNI | NULL    |                |
| title            | varchar(255) | NO   |     | NULL    |                |
| company          | varchar(255) | NO   |     | NULL    |                |
| address          | text         | YES  |     | NULL    |                |
| due_days         | int(11)      | NO   |     | NULL    |                |
| vat              | double       | YES  |     | 0       |                |
| calculator       | varchar(20)  | NO   |     | NULL    |                |
| number_generator | varchar(20)  | NO   |     | NULL    |                |
| renderer         | varchar(20)  | NO   |     | NULL    |                |
| payment_terms    | text         | YES  |     | NULL    |                |
| vat_id           | varchar(50)  | YES  |     | NULL    |                |
| contact          | longtext     | YES  |     | NULL    |                |
| payment_details  | longtext     | YES  |     | NULL    |                |
| language         | varchar(6)   | NO   |     | NULL    |                |
+------------------+--------------+------+-----+---------+----------------+
```
decimal_duration removed and not null on language is set.
For the deletes several seems to remain though.
```
select * from kimai2_user_preferences where name in ('reporting.initial_view','hours_24','theme.collapsed_sidebar','collapsed_sidebar','theme.layout','theme_layout','layout');
+----+---------+-------------------------+--------------+
| id | user_id | name                    | value        |
+----+---------+-------------------------+--------------+
| 23 |       1 | hours_24                | 1            |
| 24 |       1 | theme.layout            | fixed        |
| 25 |       1 | theme.collapsed_sidebar |              |
| 26 |       1 | reporting.initial_view  | week_by_user |
+----+---------+-------------------------+--------------+
4 rows in set (0.000 sec)
```
When I ran selects for the remaining delete statements I got no results. Should I manually delete those rows from kimai2_user_preferences?

**Reply by kevinpapst** [reply]
Doesn't matter. Your DB doesn't care about some unused rows 😁 
But these preferences are not read anymore, so removing them won't hurt either.
